### DISCUSSION TITLE
Origin is not allowed by Access-Control-Allow-Origin. Status code: 204

### DISCUSSION THREAD

**Discussion by giuliomagnifico** [umami-software/umami#1366] [Q&A]
Hi, I found my trouble to my previous issue (https://github.com/umami-software/umami/issues/1355), umami doesn't collect data because, well I don't know, but I get these errors:

```
XMLHttpRequest cannot load https://umami.giuliomagnifico.blog/api/collect due to access control checks.


Origin https://giuliomagnifico.blog is not allowed by Access-Control-Allow-Origin. Status code: 204 https://umami.giuliomagnifico.blog/api/collect
```

What I tried is to enable the domain (subdomain), still get the errors and no data, so I enabled the https acess (I thought was a certificate issue), but the same, then I found [this](https://github.com/umami-software/umami/issues/648) discussion and I added to my nginx:
`   proxy_hide_header 'Access-Control-Allow-Origin';`

But I still get the errors and data are not collected in umami. There's a way to fix it? 
Thanks

**Comment by mikecao** [comment]
Your website probably defined some CORS headers that is preventing external calls. See https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

**Reply by giuliomagnifico** [reply]
Thanks for the reply, finally I found the issue always following the discussion [already linked](https://github.com/umami-software/umami/issues/648), I wrote badly a line, now it works as it should.

This fixed it: https://github.com/umami-software/umami/issues/648#issuecomment-907749879

(Maybe you can plan to link the discussion or the fix for nginx in the FAQ or somewhere in the tutorials).

You can close and mark as solved the discussion.
